src/docfx/restore/RestoreGitMap.cs

```diff
@@ -11,67 +11,77 @@ namespace Microsoft.Docs.Build
 {
     internal class RestoreGitMap : IDisposable
     {
+        private readonly string _docsetPath;
+        private readonly IReadOnlyDictionary<PackageUrl, DependencyGitLock> _dependencyGitLock;
         private readonly IReadOnlyDictionary<(PackageUrl packageUrl, string commit), (string path, DependencyGit git)> _acquiredGits;
 
-        public Dictionary<PackageUrl, DependencyGitLock> DependencyGitLock { get; private set; }
-
-        public RestoreGitMap(IReadOnlyDictionary<(PackageUrl packageUrl, string commit), (string path, DependencyGit git)> acquiredGits = null)
+        public RestoreGitMap(
+            string docsetPath,
+            IReadOnlyDictionary<PackageUrl, DependencyGitLock> dependencyGitLock,
+            IReadOnlyDictionary<(PackageUrl packageUrl, string commit), (string path, DependencyGit git)> acquiredGits)
         {
-            _acquiredGits = acquiredGits ?? new Dictionary<(PackageUrl packageUrl, string commit), (string path, DependencyGit git)>();
+            _docsetPath = docsetPath;
+            _dependencyGitLock = dependencyGitLock;
+            _acquiredGits = acquiredGits;
         }
 
-        /// <summary>
-        /// The dependency lock must be loaded before using this method
-        /// </summary>
-        public string GetGitRestorePath(PackageUrl url, string docsetPath)
+        public string GetGitRestorePath(PackageUrl packageUrl)
         {
-            switch (url.Type)
+            var path = TryGetGitRestorePath(packageUrl);
+            if (path != null)
             {
-                case PackageType.Folder:
-                    var fullPath = Path.Combine(docsetPath, url.Path);
-                    if (Directory.Exists(fullPath))
-                    {
-                        return fullPath;
-                    }
+                return path;
+            }
 
+            switch (packageUrl.Type)
+            {
+                case PackageType.Folder:
                     // TODO: Intentionally don't fallback to fallbackDocset for git restore path,
                     // TODO: populate source info
-                    throw Errors.FileNotFound(new SourceInfo<string>(url.Path)).ToException();
+                    throw Errors.FileNotFound(new SourceInfo<string>(packageUrl.Path)).ToException();
 
                 case PackageType.Git:
-                    return GetGitRestorePath(url);
+                    throw Errors.NeedRestore(packageUrl.ToString()).ToException();
 
                 default:
-                    throw new NotSupportedException($"Unknown package url: '{url}'");
+                    throw new NotSupportedException($"Unknown package url: '{packageUrl}'");
             }
         }
 
-        /// <summary>
-        /// The dependency lock must be loaded before using this method
-        /// </summary>
-        public string GetGitRestorePath(PackageUrl packageUrl)
+        public string TryGetGitRestorePath(PackageUrl packageUrl)
```

Comment:
Then we have two methods with same name "TryGetRestorePath".

With Dependency Lock, we should avoid to provide TryXXX because every package should map to a restored path, otherwise we throw need-restore errors.